### PR TITLE
Extend client to combine offers

### PR DESCRIPTION
**Pull request by papagian**
This commit introduces a new client method which combines multiple
offers in one ACCEPT request.
This is useful for facing the agent resource fragmentation.

### CODE REVIEW

File: mesoshttp/client.py [hunk 1]
```diff
@@ -713,3 +713,67 @@ def __register(self):
...
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+            'Mesos-Stream-Id': self.streamId
+        }
+
+        tasks = []
+        for operation in operations:
+            if 'agent_id' not in operation:
+                operation['agent_id'] = {
+                    'value': self.offer[0]['agent_id']['value']
```
Comment (osallou) [inline mesoshttp/client.py]: What is this self.offer here? I do not see any reference to this.
In client, this is no offer instance attribute

Comment (papagian) [reply mesoshttp/client.py]: You are right. I will fix it.

File: mesoshttp/client.py [hunk 2]
```diff
@@ -713,3 +713,67 @@ def __register(self):
...
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+            'Mesos-Stream-Id': self.streamId
+        }
+
+        tasks = []
+        for operation in operations:
+            if 'agent_id' not in operation:
+                operation['agent_id'] = {
+                    'value': ids[0]
```
Comment (osallou) [inline mesoshttp/client.py]: You changed operation agent id to ids[0] but this refers to an offer id, not an agent id.
And each operation may have a different agent id if i am correct, forcing to first offer may not be good depending on how mesos manage it.

Merging multiple offers is not so easy as it seems ;-)

Comment (papagian) [reply mesoshttp/client.py]: Oops, you are right about the ids variable.

Concerning the other, the [Scheduler API doc](https://mesos.apache.org/documentation/latest/scheduler-http-api/#accept) is not very clear about whether it is possible to combine offers of different agents. However, in my case I only need to combine offers of the same agent.

I think a better approach would be to leave each operation intact (as it is passed) and let Mesos to process the request and complain if its invalid. Therefore I would remove the whole for loop.
What do you think?

**Review by osallou — COMMENTED** [review]
Considering each operation is 'correct' could  indeed be enough, not managing agent id, supposing it is already set in operation.

**Comment by osallou** [comment]
You should however specify this constraint in method docstring

**Pull request merged**